src/platform/forms-system/src/js/containers/FormPage.jsx

```diff
@@ -110,6 +113,38 @@ class FormPage extends React.Component {
     this.props.router.push(path);
   };
 
+  prePopulateArrayData = () => {
+    const { pageConfig } = this.props.route;
+    // only applicable to array routes with these settings
+    if (pageConfig.showPagePerItem && pageConfig.allowPathWithNoItems) {
+      const arrayFormData = get(
+        [pageConfig.arrayPath, this.props.params.index],
+        this.props.form.data,
+      );
+      if (!arrayFormData) {
```

Comment:
This won't work if the `arrayFormData` is an empty object, right? Does the data really need to be prepopulated?

Reply:
An empty object is OK, but one of the problems arises when you have an object inside of an object, like address pattern is encapsulated inside an object. Then suddenly all the logic that scans for updateSchema, required fields etc, will not be able to find its target since it does a lot of stuff based on schema vs formData in `forms-system/.../state/helpers` , and likely some other files. updateSchemas won't trigger, requireds won't find their targets, page number won't load properly (no index in the array).

onChange's solve the problem because once a field has changed, then suddenly that item will have data (assuming it didn't have data before), and magically everything works. but onChange doesn't happen on initial page load, which is why updating (prepopulating) data onMount/onUpdate seems like the next logical place.

Ideally maybe I could update all the helpers and various files and internals, but there are so many things that break without, and are fixed from this one change, that this seems like a reasonable solution.